### ISSUE TITLE
MinGW CI build needs fix to work around GCC 12 warning

### ISSUE THREAD

**Issue by randombit** [randombit/botan#3709]
GCC 12 introduces some bogus warnings in STL usage, and it looks like the MinGW GCC has upgraded so that CI build is breaking due to use of `-Werror`.

```
2023-09-21T18:18:57.6966905Z     inlined from 'constexpr void std::sort(_RAIter, _RAIter) [with _RAIter = __gnu_cxx::__normal_iterator<vector<unsigned char, Botan::secure_allocator<unsigned char> >*, vector<vector<unsigned char, Botan::secure_allocator<unsigned char> > > >]' at C:/ProgramData/Chocolatey/lib/mingw/tools/install/mingw64/lib/gcc/x86_64-w64-mingw32/12.2.0/include/c++/bits/stl_algo.h:4820:18,
2023-09-21T18:18:57.6968126Z     inlined from 'void Botan::DER_Encoder::DER_Sequence::push_contents(Botan::DER_Encoder&)' at src/lib/asn1/der_enc.cpp:82:16:
2023-09-21T18:18:57.6969516Z C:/ProgramData/Chocolatey/lib/mingw/tools/install/mingw64/lib/gcc/x86_64-w64-mingw32/12.2.0/include/c++/bits/stl_algobase.h:1843:41: error: 'int __builtin_memcmp(const void*, const void*, long long unsigned int)' specified bound [9223372036854775808, 18446744073709551615] exceeds maximum object size 9223372036854775807 [-Werror=stringop-overread]
2023-09-21T18:18:57.6970353Z  1843 |                       = __builtin_memcmp(&*__first1, &*__first2, __len) <=> 0;
2023-09-21T18:18:57.6970739Z       |                         ~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

working around this is complicated by the fact that

- The warnings were introduced in GCC 12
- GCC rejects `-Wno-foo` if that version does not know about `-Wfoo` (which always seemed stupid to me, but that's how it works)

So we can't just use `-Wno-error=foo` since that breaks earlier GCC.

The short term fix would just be to disable use of `-Werror` in the MinGW build but it would be nice to handle this more generically.

**Comment by randombit** [comment]
Ah it appears `-Wstringop-overread` is in GCC 11 so for this case I think we can just add it to the `-Wno-error` list and don't have to do anything complicated.

**Comment by randombit** [comment]
Fixed in #3711
